components/tidb_query_datatype/src/codec/data_type/mod.rs

```diff
@@ -266,7 +266,7 @@ impl Evaluable for Int {
             ScalarValue::Int(x) => x.as_ref(),
             ScalarValue::Enum(x) => x
                 .as_ref()
-                .map(|x| unsafe { std::mem::transmute(x.value_ref()) }),
+                .map(|x| unsafe { &*(x.value_ref() as *const u64 as *const i64) }),
```

Comment:
I would prefer `std::mem::transmute<&u64, &i64>(a, b)` here. The reason is that, by specifying concrete type here, we could retain the original lifetime of `&u64`. Neither `transmute` without generic and pointer cast could have the same effect.

By the way, if we always get `Enum` as `i64`, why not using i64 across all enum-related structs?

Reply:
Thanks, let me try the `transmute` you mentioned.

Reply:
I think storing `u64` could be more consistent.

Reply:
This is a recommendation from CI. CI will not be happy if I write `transmute` here😂

Reply:
How about copy here instead of using unsafe magic?

Reply:
Copied item only have lifetime throughout this function. We need it to be as long as value itself.

Reply:
OK, I got it.